packages/ui/src/models/visualization/flows/abstract-camel-visual-entity.ts

```diff
@@ -141,7 +141,7 @@ export abstract class AbstractCamelVisualEntity<T extends object> implements Bas
       /** If we're in Replace mode, we need to delete the existing step */
       const deleteCount = options.mode === AddStepMode.ReplaceStep ? 1 : 0;
 
-      const stepsArray: ProcessorDefinition[] = getValue(this.entityDef, pathArray.slice(0, -2), []);
+      const stepsArray: ProcessorDefinition[] = getArrayProperty(this.entityDef, pathArray.slice(0, -2).join('.'));
```

Comment:
This was the root cause of the bug preventing adding steps in EIPs with `steps: undefined`